### PR TITLE
feat: support nonce for starknet 0.10

### PR DESCRIPTION
**Pull request by Mirko-von-Leipzig**
This PR adds support for contract nonces in preparation for the `starknet 0.10` release.

While the file and line diff count may appear intimidating, the majority of it is replacing outdated fixtures.

Reviewing by commit should be fairly clean.

Summary of the changes:

- replace state update fixtures checking for compatibility with `< 0.9.0`
- add integration state update fixtures for `0.10`
- migration to add nonces to `contract_states` table
- connect sync and state hash calculation to nonces
- connect rpc to stored nonces

The migration was "instant" on both mainnet and goerli so I didn't add a timing warning.

I'm also trying a `FIXME(<version>)` tagging system for tracking fixme's once `starknet 0.10` is on mainnet.

### CODE REVIEW

File: crates/pathfinder/fixtures/sequencer/integration/state_update/216572.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                }
+            ]
+        },
+        "nonces": {
+            "0x577abc3e3ab491af6fdc1e185b71a6d04f7e71a525f9f57c19fc36ed0655a39": "0xbf"
+        },
+        "deployed_contracts": [],
+        "declared_contracts": []
+    }
+}
```
Comment (koivunej) [inline crates/pathfinder/fixtures/sequencer/integration/state_update/216572.json:30]: Missing \n

Comment (Mirko-von-Leipzig) [reply crates/pathfinder/fixtures/sequencer/integration/state_update/216572.json:30]: Guess my json formatter doesn't insert them :) Will fix.

File: crates/pathfinder/src/rpc/api.rs [hunk 2]
```diff
@@ -1020,7 +1020,6 @@ impl RpcApi {
     /// Not currently supported correctly as nonce's aren't implemented yet. In the mean time
     /// returns "0x0" until starknet reaches 0.10 at which point it will return an error instead.
```
Comment (kkovaacs) [inline crates/pathfinder/src/rpc/api.rs:1021]: I guess this comment is outdated now.

**Review by kkovaacs — APPROVED** [review]

**Review by CHr15F0x — APPROVED** [review]

**Review by koivunej — APPROVED** [review]
Looking at only state update related parts this looks good, didn't have time for rest.

**Pull request merged**